src/System.Console/tests/TermInfo.cs

```diff
@@ -0,0 +1,132 @@
+                }
+            }
+        }
+    }
+
+    [Theory]
+    [PlatformSpecific(PlatformID.AnyUnix)]
+    [InlineData("xterm-256color", "\u001B\u005B\u00330m", "\u001B\u005B\u00340m", 0)]
+    [InlineData("xterm-256color", "\u001B\u005B\u00331m", "\u001B\u005B\u00341m", 1)]
+    [InlineData("xterm-256color", "\u001B\u005B\u00338m", "\u001B\u005B\u00348m", 8)]
+    [InlineData("screen", "\u001B\u005B\u00330m", "\u001B\u005B\u00340m", 0)]
+    [InlineData("screen", "\u001B\u005B\u00332m", "\u001B\u005B\u00342m", 2)]
+    [InlineData("screen", "\u001B\u005B\u00339m", "\u001B\u005B\u00349m", 9)]
+    [InlineData("Eterm", "\u001B\u005B\u00330m", "\u001B\u005B\u00340m", 0)]
+    [InlineData("Eterm", "\u001B\u005B\u00333m", "\u001B\u005B\u00343m", 3)]
+    [InlineData("Eterm", "\u001B\u005B\u003310m", "\u001B\u005B\u003410m", 10)]
+    [InlineData("wsvt25", "\u001B\u005B\u00330m", "\u001B\u005B\u00340m", 0)]
+    [InlineData("wsvt25", "\u001B\u005B\u00334m", "\u001B\u005B\u00344m", 4)]
+    [InlineData("wsvt25", "\u001B\u005B\u003311m", "\u001B\u005B\u003411m", 11)]
+    [InlineData("mach-color", "\u001B\u005B\u00330m", "\u001B\u005B\u00340m", 0)]
+    [InlineData("mach-color", "\u001B\u005B\u00335m", "\u001B\u005B\u00345m", 5)]
+    [InlineData("mach-color", "\u001B\u005B\u003312m", "\u001B\u005B\u003412m", 12)]
+    public void TermInfoVerification(string termToTest, string expectedForeground, string expectedBackground, int colorValue)
+    {
+        object info = CreateTermColorInfo(ReadTermInfoDatabase(termToTest));
+        Assert.Equal(expectedForeground, EvaluateParameterizedStrings(GetForegroundFormat(info), colorValue));
+        Assert.Equal(expectedBackground, EvaluateParameterizedStrings(GetBackgroundFormat(info), colorValue));
+    }
+
+    [Fact]
+    [PlatformSpecific(PlatformID.OSX)]
+    public void EmuTermInfoDoesntBreakParser()
+    {
+        // This file (available by default on OS X) is called out specifically since it contains a format where it has %i
+        // but only one variable instead of two. Make sure we don't break in this case
+        TermInfoVerification("emu", "\u001br1", "\u001bs1", 0);
+    }
+
+    [Fact]
+    [PlatformSpecific(PlatformID.AnyUnix)]
+    public void TryingToLoadTermThatDoesNotExistDoesNotThrow()
+    {
+        object db = ReadTermInfoDatabase("foobar____");
+        object info = CreateTermColorInfo(db);
+        Assert.Null(db);
+        Assert.NotNull(GetBackgroundFormat(info));
+        Assert.NotNull(GetForegroundFormat(info));
+        Assert.NotNull(GetMaxColors(info));
+        Assert.NotNull(GetResetFormat(info));
+    }
+
+    private object ReadTermInfoDatabase(string term)
+    {
+        MethodInfo readDbMethod = Type.GetType("ConsolePal.TermInfo.Database").GetTypeInfo().DeclaredMethods.Where(m => m.Name == "ReadDatabase").Where(m1 => m1.GetParameters().Count() == 1).Select(m2 => m2).Single();
```

Comment:
Rather than iterating through DeclaredMethods, can you just use GetDeclaredMethod?